### PR TITLE
Operator Api | Introduce product validations

### PR DESCRIPTION
**Pull request by tom-ioki**
This PR will introduce the ProductValidation model and a custom endpoint `product_validations` that allows fetching all validations for a specific product.

### CODE REVIEW

File: lib/ioki/model/operator/product_bulletin.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                  on:   :read,
+                  type: :string
+
+        attribute :description,
+                  on:   :read,
+                  type: :string
+
+        attribute :affected_items,
+                  on:         :read,
+                  type:       :array,
```
Comment (jai-ioki) [inline lib/ioki/model/operator/product_bulletin.rb:29]: This is probably intentional but it startled me for a moment:

When there are no affected items like for `not_live` the API returns `null`

```json
  {
        "id": "not_live",
        "severity": "danger",
        "title": "Product NOT live",
        "description": "This product is in \"planned\" mode. If you operate a service, make sure to set this product to \"live\" mode, as non live products skip safety checks and critical errors might not get detected.",
        "affected_items": null,
        "type": "product_bulletin"
      }
```

When reading the attribute definition above I would expect a conversion, however its `nil`

```ruby
{:type=>"product_bulletin",
 :id=>"not_live",
 :severity=>"danger",
 :title=>"Produkt NICHT Live",
 :description=>
  "Dieses Produkt befindet sich im Modus \"planned\". Wenn Sie einen Dienst betreiben, stellen Sie sicher, dass dieses Produkt in den \"Live\"-Modus versetzt wird, da Produkte, die nicht Live sind, Sicherheitsprüfungen überspringen und kritische Fehler möglicherweise nicht erkannt werden.",
 :affected_items=>nil}
```

**Review by jai-ioki — APPROVED** [review]
Tested this locally. Looks good. I just have a small remark.

File: lib/ioki/model/operator/product_bulletin.rb (hunk 1 above)
Comment (tom-ioki) [reply lib/ioki/model/operator/product_bulletin.rb:29]: You mean, you'd expect an empty hash for `affected_items`?

Comment (jai-ioki) [reply lib/ioki/model/operator/product_bulletin.rb:29]: An empty array, yes

```
{:type=>"product_bulletin", :id=>"not_live", :severity=>"danger", :title=>"Produkt NICHT Live", :description=> "Dieses Produkt befindet sich im Modus \"planned\". Wenn Sie einen Dienst betreiben, stellen Sie sicher, dass dieses Produkt in den \"Live\"-Modus versetzt wird, da Produkte, die nicht Live sind, Sicherheitsprüfungen überspringen und kritische Fehler möglicherweise nicht erkannt werden.", :affected_items=>[]}
```

Comment (tom-ioki) [reply lib/ioki/model/operator/product_bulletin.rb:29]: Good point!
The issue can be found in the `ProductBulletin` class in triebwerk. The default value for affected_items is `nil` 😬 . I will make a follow up PR in triebwerk. 👍

Comment (tom-ioki) [reply lib/ioki/model/operator/product_bulletin.rb:29]: Let's change it to an empty array, that's more consistent.

**Pull request merged**